awscli/customizations/iot.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""
+This customization makes it easier to save various pieces of data
+returned from iot commands that would typically need to be saved to a
+file. This customization adds the following options:
+
+- aws iot create-certificate-from-csr
+  - ``--certificate-pem-outfile``: certificatePem
+- aws iot create-keys-and-certificate
+  - ``--certificate-pem-outfile``: certificatePem
+  - ``--public-key-outfile``: keyPair.PublicKey
+  - ``--private-key-outfile``: keyPair.PrivateKey
+"""
+from awscli.customizations.arguments import QueryOutFileArgument
+
+
+class IoTWrapper(object):
+
+    def __init__(self, event_handler):
+        self._cert_arg = QueryOutFileArgument(
+            'certificate-pem-outfile', 'certificatePem')
+        self._pub_key_arg = QueryOutFileArgument(
+            'public-key-outfile', 'keyPair.PublicKey')
+        self._private_key_arg = QueryOutFileArgument(
+            'private-key-outfile', 'keyPair.PrivateKey')
+        event_handler.register(
+            'building-argument-table.iot.create-keys-and-certificate',
+            self._add_create_keys_and_cert_args)
+        event_handler.register(
+            'building-argument-table.iot.create-certificate-from-csr',
+            self._add_create_cert_from_csr_args)
+        save_events = ['after-call.iot.CreateCertificateFromCsr',
+                       'after-call.iot.CreateKeysAndCertificate']
+        for event in save_events:
```

Comment:
It seems sort of weird that we are registering these events prior to we actually knowing that the arguments are present on the command line. I sort of like the pattern we do for ``streamingoutputargs.py`` where we register the event when we add it to the params.